tasks/coverage/src/sourcemap.rs

```diff
@@ -0,0 +1,136 @@
+        !path.contains("input.js")
+    }
+
+    fn run_coverage(&self, name: &str, _args: &crate::AppArgs) {
+        let path = project_root().join(format!("tasks/coverage/{name}.snap"));
+        let mut file = File::create(path).unwrap();
+
+        let mut tests = self.get_test_cases().iter().collect::<Vec<_>>();
+        tests.sort_by_key(|case| case.path());
+
+        for case in tests {
+            let result = case.test_result();
+            let path = case.path().to_string_lossy();
+            let result = match result {
+                TestResult::Snapshot(snapshot) => snapshot,
+                TestResult::ParseError(error, _) => error,
+                _ => {
+                    unreachable!()
+                }
+            };
+            writeln!(file, "- {path}").unwrap();
+            writeln!(file, "{result}\n\n").unwrap();
+        }
+    }
+}
+
+pub struct SourcemapCase {
+    path: PathBuf,
+    code: String,
+    source_type: SourceType,
+    result: TestResult,
+}
+
+impl SourcemapCase {
+    pub fn source_type(&self) -> SourceType {
+        self.source_type
+    }
+
+    fn create_visualizer_url(code: &str, map: &str) -> String {
+        let hash =
+            BASE64_STANDARD.encode(format!("{}\0{}{}\0{}", code.len(), code, map.len(), map));
+        format!("https://evanw.github.io/source-map-visualization/#{hash}")
+    }
+}
+
+impl Case for SourcemapCase {
+    fn new(path: PathBuf, code: String) -> Self {
+        let source_type = SourceType::from_path(&path).unwrap();
+        Self { path, code, source_type, result: TestResult::ToBeRun }
+    }
+
+    fn code(&self) -> &str {
+        &self.code
+    }
+
+    fn path(&self) -> &Path {
+        &self.path
+    }
+
+    fn test_result(&self) -> &TestResult {
+        &self.result
+    }
+
+    fn run(&mut self) {
+        let source_type = self.source_type();
+        self.result = self.execute(source_type);
+    }
+
+    fn execute(&mut self, source_type: SourceType) -> TestResult {
+        let source_text = self.code();
+        let allocator = oxc_allocator::Allocator::default();
+        let ret = oxc_parser::Parser::new(&allocator, source_text, source_type).parse();
+
+        if !ret.errors.is_empty() {
+            if let Some(error) = ret.errors.into_iter().next() {
+                let error = error.with_source_code(source_text.to_string());
+                return TestResult::ParseError(error.to_string(), false);
+            }
+        }
+
+        let codegen_options = oxc_codegen::CodegenOptions::default();
+        let (content, map) = oxc_codegen::Codegen::<false>::new(source_text.len(), codegen_options)
+            .build_with_sourcemap(&ret.program, source_text, "");
+        let mut buff = vec![];
+        map.to_writer(&mut buff).unwrap();
+
+        let mut result = String::from_utf8(buff).unwrap();
+        result.push('\n');
+        result.push_str(Self::create_visualizer_url(&content, &result).as_str());
```

Comment:
I think we should discuss how to test this, this is just printing things out, there is no validation yet :-(